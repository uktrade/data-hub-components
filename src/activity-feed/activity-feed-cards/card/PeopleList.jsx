import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Link } from 'govuk-react'

const UList = styled('ul')`
  margin: 0;
  padding-left: 0
`

const ListItem = styled('li')`
  list-style-type: none;
`

const Person = ({ person }) => {
  const name = person.url ? <Link href={person.url}>{person.name}</Link> : person.name
  const emailAddress = <Link href={`mailto:${person.emailAddress}`}> {person.emailAddress}</Link>
  const jobTitle = person.jobTitle ? <span>({person.jobTitle})</span> : null
  const team = person.team ? <span>({person.team})</span> : null

  return <ListItem>{name} {emailAddress} {jobTitle} {team}</ListItem>
}

Person.propTypes = {
  person: PropTypes.object.isRequired,
}

const PeopleList = ({ people }) => {
  return <UList>{ people.map(person => <Person key={person.id} person={person} />) }</UList>
}

PeopleList.propTypes = {
  people: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    emailAddress: PropTypes.string,
    jobTitle: PropTypes.string,
  })).isRequired,
}

export default PeopleList
