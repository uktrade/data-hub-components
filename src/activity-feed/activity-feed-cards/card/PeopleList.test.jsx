import React from 'react'
import renderer from 'react-test-renderer'

import PeopleList from './PeopleList'

describe('PeopleList.js', () => {
  describe('when the list of people is empty', () => {
    test('should not render an empty list', () => {
      const people = []
      const tree = renderer.create(<PeopleList people={people} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
