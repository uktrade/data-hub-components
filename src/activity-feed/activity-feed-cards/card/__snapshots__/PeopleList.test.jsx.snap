// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleList.js when the list of people is empty should not render an empty list 1`] = `
<ul
  className="sc-cmthru hkmdGA"
/>
`;
